connection.go

```diff
@@ -468,9 +480,7 @@ func (c *Connection) sendUnflushed(f frame) error {
 		return ErrClosed
 	}
 
-	c.sendM.Lock()
-	err := f.write(c.writer.w)
-	c.sendM.Unlock()
+	err := c.writer.WriteFrameNoFlush(f)
```

Comment:
Don't you need a lock here to prevent concurrent access to `Connection.writer` here?

Reply:
No, go up a function call and you'll see that this is only called within a locked section of code.